ducktape/tests/result.py

```diff
@@ -69,59 +81,141 @@ def __repr__(self):
+            "function_name": self.function_name,
+            "injected_args": self.injected_args,
+            "description": self.description,
+            "results_dir": self.results_dir,
+            "relative_results_dir": self.relative_results_dir,
+            "base_results_dir": self.base_results_dir,
+            "test_status": self.test_status,
+            "summary": self.summary,
+            "data": self.data,
+            "start_time": self.start_time,
+            "stop_time": self.stop_time,
+            "run_time_seconds": self.run_time_seconds,
+            "nodes_allocated": self.nodes_allocated,
+            "nodes_used": self.nodes_used,
+            "services": self.services
+        }
+
+
+class TestResults(object):
     """Class used to aggregate individual TestResult objects from many tests."""
-    # TODO make this tread safe - once tests are run in parallel, this will be shared by multiple threads
-
-    def __init__(self, session_context):
+    def __init__(self, session_context, cluster):
         """
         :type session_context: ducktape.tests.session.SessionContext
         """
-        super(list, self).__init__()
-
+        self._results = []
         self.session_context = session_context
+        self.cluster = cluster
 
         # For tracking total run time
         self.start_time = -1
         self.stop_time = -1
 
+    def append(self, obj):
+        return self._results.append(obj)
+
+    def __len__(self):
+        return len(self._results)
+
+    def __iter__(self):
+        return iter(self._results)
+
     @property
     def num_passed(self):
-        return len([r for r in self if r.test_status == PASS])
+        return len([r for r in self._results if r.test_status == PASS])
 
     @property
     def num_failed(self):
-        return len([r for r in self if r.test_status == FAIL])
+        return len([r for r in self._results if r.test_status == FAIL])
 
     @property
     def num_ignored(self):
-        return len([r for r in self if r.test_status == IGNORE])
+        return len([r for r in self._results if r.test_status == IGNORE])
 
     @property
-    def run_time(self):
+    def run_time_seconds(self):
         if self.start_time < 0:
             return -1
         if self.stop_time < 0:
-            return time.time() - self.start_time
+            self.stop_time = time.time()
 
         return self.stop_time - self.start_time
 
     def get_aggregate_success(self):
         """Check cumulative success of all tests run so far
         :rtype: bool
         """
-        for result in self:
+        for result in self._results:
             if result.test_status == FAIL:
                 return False
         return True
 
-
+    def _stats(self, num_list):
+        if len(num_list) == 0:
+            return {
+                "mean": None,
+                "min": None,
+                "max": None
+            }
+
+        return {
+            "mean": statistics.mean(num_list),
```

Comment:
Fine to keep it in here, but including a whole package for something that can be done as `sum(x)/float(len(x))` feels like a bit of overkill.